modules/functional_expansion_tools/src/functions/FunctionSeries.C

```diff
@@ -69,7 +80,9 @@ FunctionSeries::FunctionSeries(const InputParameters & parameters)
     _x(getParam<MooseEnum>("x")),
     _y(getParam<MooseEnum>("y")),
     _z(getParam<MooseEnum>("z")),
-    _disc(getParam<MooseEnum>("disc"))
+    _disc(getParam<MooseEnum>("disc")),
+    _expansion_type(getParam<MooseEnum>("expansion_type")),
+    _generation_type(getParam<MooseEnum>("generation_type"))
```

Comment:
Since you went through the trouble of creating corresponding C enums for these MooseEnums you should probably just convert them right here where you pull them out and avoid using the MooseEnums after they've been parsed and validated. The C enums are more type-safe and faster when used in hotspots.

Reply:
To convert these would require machinery like that in `Conversion` right? Too me that seems like too much work; I would rather just change the `switch, case` to chained `else if`s and use the original `MooseEnum`

Reply:
Alternatively, I could do something like`_expansion_type(static_cast<NormalizationType>(getParam<MooseEnum>("expansion_type").operator int()))` where `NormalizationType` is the name of the `enum`. 😄 However, I think I'll stick with the `MooseEnum` throughout

Reply:
More like this: `getParam<MooseEnum>("expansion_type").getEnum<ExpansionEnum>()`